Sources/WASI/FileSystem.swift

```diff
@@ -120,3 +127,55 @@ struct FdTable {
         }
     }
 }
+
+/// Content of a file that can be retrieved from the file system.
+public enum FileContent {
+    case bytes([UInt8])
+    case handle(FileDescriptor)
+}
+
+/// Public protocol for file system providers that users interact with.
+///
+/// This protocol exposes only user-facing methods for managing files and directories.
+public protocol FileSystemProvider {
+    /// Adds a file to the file system with the given byte content.
+    func addFile(at path: String, content: [UInt8]) throws
```

Comment:
Let's not hardcode byte arrays here, but allow more types, like `ArraySlice<UInt8>`, `ByteBufferView`, `UnsafeBufferPointer<UInt8>` etc.

```suggestion
    func addFile(at path: String, content: some Sequence<UInt8>) throws
```